porcupine/plugins/python_venv.py

```diff
@@ -21,14 +21,18 @@
 
 
 def is_venv(path: Path) -> bool:
-    landmarks = [path / "pyvenv.cfg"]
-    if sys.platform == "win32":
-        landmarks.append(path / "Scripts" / "python.exe")
-        landmarks.append(path / "Scripts" / "activate.bat")
+    if not "porcupine-venv" in str(path):
+        print(f"{path} is valid venv")
+        landmarks = [path / "pyvenv.cfg"]
+        if sys.platform == "win32":
+            landmarks.append(path / "Scripts" / "python.exe")
+            landmarks.append(path / "Scripts" / "activate.bat")
+        else:
+            landmarks.append(path / "bin" / "python3")
+            landmarks.append(path / "bin" / "activate")
+        return all(landmark.exists() for landmark in landmarks)
     else:
-        landmarks.append(path / "bin" / "python3")
-        landmarks.append(path / "bin" / "activate")
-    return all(landmark.exists() for landmark in landmarks)
+        log.info(f"not using venv: '{path}', as it contains 'porcupine-venv'")
```

Comment:
This branch needs a `return False` at the end to make the type checker happy. It will return `None` by default, which acts as a falsy value, but `return False` makes it obvious how this works.